lib/gui/components/drive-selector/templates/drive-selector-modal.tpl.html

```diff
@@ -15,35 +15,14 @@ <h4 class="list-group-item-heading">{{ drive.description }} - {{ drive.size | gi
 
           <footer class="list-group-item-footer">
 
-            <span class="label label-warning"
-                ng-show="modal.constraints.isDriveLargeEnough(drive, modal.state.getImage()) && !modal.constraints.isDriveLocked(drive) && !modal.constraints.isDriveSizeRecommended(drive, modal.state.getImage())">
-                <i class="glyphicon glyphicon-warning-sign"></i>
-                Not Recommended</span>
-
-            <!-- There can be a case where the device it not large enough, and it's also locked. -->
-            <!-- Since in this case both labels will be displayed, we chose to only show the -->
-            <!-- "not large enough label", since from the point of view of the user, the locked -->
-            <!-- state is irrelevant if the drive is not large enough anyway. -->
-
-            <span class="label label-danger"
-              ng-hide="modal.constraints.isDriveLargeEnough(drive, modal.state.getImage())">
-                <i class="glyphicon glyphicon-resize-small"></i>
-                Too Small For Image</span>
-
-            <span class="label label-danger"
-              ng-show="modal.constraints.isDriveLocked(drive) && modal.constraints.isDriveLargeEnough(drive, modal.state.getImage())">
-                <i class="glyphicon glyphicon-lock"></i>
-                Locked</span>
-
-            <span class="label label-danger"
-              ng-show="modal.constraints.isSystemDrive(drive)">
-                <i class="glyphicon glyphicon-hdd"></i>
-                System Drive</span>
-
-            <span class="label label-danger"
-              ng-show="modal.constraints.isSourceDrive(drive, modal.state.getImage())">
-                <i class="glyphicon glyphicon-circle-arrow-down"></i>
-                Source Drive</span>
+            <span class="label"
+              ng-class="{
+                'label-warning': modal.constraints.getDriveFeedback(drive, modal.state.getImage()).type === 'warning',
+                'label-danger': modal.constraints.getDriveFeedback(drive, modal.state.getImage()).type === 'error'
+              }"
+              ng-show="modal.constraints.getDriveFeedback(drive, modal.state.getImage()).message">
+                <b>{{ modal.constraints.getDriveFeedback(drive, modal.state.getImage()).message }}</b>
```

Comment:
If the same function is being called 4 times, should the `.getDriveFeedback` function be memoized?